### PR TITLE
nghttp2: upgrade version 1.57.0 -> 1.61.0 to address CVE-2024-28182

### PR DESCRIPTION
**Pull request by mfrw**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
- **nghttp2: upgrade version 1.59.0 -> 1.61.0 to address CVE-2024-28182**

###### Change Log  

- **nghttp2: upgrade version 1.59.0 -> 1.61.0 to address CVE-2024-28182**

###### Does this affect the toolchain?  

**YES**

###### Associated issues  

- #10064

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2024-28182

###### Test Methodology

- Local Build
- Pipeline build id: [PR-10076](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=618721&view=results)

### CODE REVIEW

File: SPECS/nghttp2/nghttp2.spec [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Summary:        nghttp2 is an implementation of HTTP/2 and its header compression algorithm, HPACK.
 Name:           nghttp2
-Version:        1.57.0
```
Comment (jslobodzian) [inline SPECS/nghttp2/nghttp2.spec:3]: How confident are you in this version bump?  Are there any compatibility concerns?

Comment (jslobodzian) [reply SPECS/nghttp2/nghttp2.spec:3]: @mfrw kind ping?

Comment (mfrw) [reply SPECS/nghttp2/nghttp2.spec:3]: I am slightly iffy about this version bumper given what we experienced with git & git-lfs lately.

Comment (mfrw) [reply SPECS/nghttp2/nghttp2.spec:3]: Superseded by:
- https://github.com/microsoft/azurelinux/pull/10656

**Pull request closed without merging**